internal/handlers/sqlite/msg_explain.go

```diff
@@ -106,11 +106,17 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 	}
 
 	// Limit pushdown is not applied if:
-	//  - `filter` is set, it must fetch all documents to filter them in memory;
-	//  - `sort` is set but `EnableSortPushdown` is not set, it must fetch all documents
-	//  and sort them in memory;
+	//  - `filter` is set but can not be applied pushdown entirely,
+	//  or `filter` is set and `DisableFilterPushdown` is set,
+	//  it must fetch all documents to filter them in memory;
+	//  - `sort` is set but can not be applied pushdown entirely,
+	//  or `sort` is set but `EnableSortPushdown` is not set,
+	//  it must fetch all documents and sort them in memory;
 	//  - `skip` is non-zero value, skip pushdown is not supported yet.
-	if params.Filter.Len() == 0 && (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
+	// More checks are performed when preparing LIMIT clause.
+	if (params.Filter.Len() == 0 || !h.DisableFilterPushdown) &&
+		(params.Sort.Len() == 0 || (h.EnableSortPushdown && params.Sort.Len() == 1)) &&
+		params.Skip == 0 {
```

Comment:
This logic became a bit too complex. Although there is comment, it doesn't quite help understanding what is going on at first sight. Could we simplify it or refactor it? :thinking:

Reply:
Quite agree. But this logic may be difficult to simplify, as it does indeed involve many parameters and flags. Luckily, with future work, logic here should be simplified, such as we won't need to check if skip is 0 if we support skip pushdown. Or, do you think it's a good idea to put logic here in one or two functions, like `if canApplyLimitPushdown(params,h){`?